src/native/crashtracker.rs

```diff
@@ -288,3 +308,140 @@ pub fn crashtracker_status() -> anyhow::Result<CrashtrackerStatus> {
 pub fn crashtracker_receiver() -> anyhow::Result<()> {
     libdd_crashtracker::receiver_entry_point_stdin()
 }
+
+/// Result type for runtime callback operations
+#[pyclass(
+    eq,
+    eq_int,
+    name = "CallbackResult",
+    module = "datadog.internal._native"
+)]
+#[derive(Debug, PartialEq, Eq)]
+pub enum CallbackResult {
+    Ok,
+    Error,
+}
+
+impl From<CallbackError> for CallbackResult {
+    fn from(_error: CallbackError) -> Self {
+        CallbackResult::Error
+    }
+}
+
+const MAX_TRACEBACK_SIZE: usize = 8 * 1024; // 8KB
+
+// Attempt to resolve _Py_DumpTracebackThreads at runtime
+// Returns None if symbol is not available
+unsafe fn get_dump_traceback_fn() -> Option<PyDumpTracebackThreadsFn> {
+    #[cfg(unix)]
+    {
+        extern "C" {
+            fn dlsym(
+                handle: *mut std::ffi::c_void,
+                symbol: *const std::ffi::c_char,
+            ) -> *mut std::ffi::c_void;
+        }
+
+        const RTLD_DEFAULT: *mut std::ffi::c_void = ptr::null_mut();
+
+        let symbol_ptr = dlsym(
+            RTLD_DEFAULT,
+            b"_Py_DumpTracebackThreads\0".as_ptr() as *const std::ffi::c_char,
+        );
+
+        if symbol_ptr.is_null() {
+            None
+        } else {
+            Some(std::mem::transmute(symbol_ptr))
+        }
+    }
+
+    #[cfg(not(unix))]
+    {
+        None
+    }
+}
+
+unsafe fn dump_python_traceback_as_string(
+    emit_stacktrace_string: unsafe extern "C" fn(*const c_char),
+) {
+    // Try to get the dump traceback function
+    let dump_fn = match get_dump_traceback_fn() {
```

Comment:
I don't see `dlsym` from https://man7.org/linux/man-pages/man7/signal-safety.7.html

I believe we don't want to call `get_dump_traceback_fn()` in here, and call it when the module or crashtracker is initialized?